ext/fileinfo/libmagic/magic.c

```diff
@@ -403,3 +403,12 @@ magic_getparam(struct magic_set *ms, int param, void *val)
 		return -1;
 	}
 }
+
+public const char *
+magic_magic_file_get(struct magic_set *ms)
+{
+	if(strncmp(ms->file, "unknown", strlen("unknown")) == 0) {
```

Comment:
I don't know how without using magic constants. This function returns the used filename and the internal database does not have a filename.